lib/Sema/TypeCheckProtocol.cpp

```diff
@@ -3664,10 +3654,10 @@ ConformanceChecker::inferTypeWitnessesViaValueWitness(ValueDecl *req,
   }
 
   auto setup = [&]() -> std::tuple<Optional<RequirementMatch>, Type, Type> {
-    fullWitnessType = removeSelfParam(witness, fullWitnessType);
+    fullWitnessType = fullWitnessType->removeSelfParam(witness);
     return std::make_tuple(
         None,
-        removeSelfParam(req, req->getInterfaceType()),
+        req->getInterfaceType()->removeSelfParam(req),
```

Comment:
I plan to use this in IDE type checking APIs, more specifically, to compare the interface types of two function decls without Self context.